sklearn/feature_selection/from_model.py

```diff
@@ -59,55 +147,131 @@ def transform(self, X, threshold=None):
+    estimator : object
+        The base estimator from which the transformer is built.
 
-                threshold = scale * reference
+    threshold : string, float or None, optional (default=None)
+            The threshold value to use for feature selection. Features whose
+            importance is greater or equal are kept while the others are
+            discarded. If "median" (resp. "mean"), then the threshold value is
+            the median (resp. the mean) of the feature importances. A scaling
+            factor (e.g., "1.25*mean") may also be used. If None and if
+            available, the object attribute ``threshold`` is used. Otherwise,
+            "mean" is used by default.
 
-            elif threshold == "median":
-                threshold = np.median(importances)
+    warm_start : bool, optional
+        When set to True, reuse the solution of the previous call to fit as
+        initialization, otherwise, just erase the previous solution.
 
-            elif threshold == "mean":
-                threshold = np.mean(importances)
+    Attributes
+    ----------
+    `estimator_`: an estimator
+        The base estimator from which the transformer is built.
 
-        else:
-            threshold = float(threshold)
+    `scores_`: array, shape=(n_features,)
+        The importance of each feature according to the fit model.
 
-        # Selection
-        try:
-            mask = importances >= threshold
-        except TypeError:
-            # Fails in Python 3.x when threshold is str;
-            # result is array of True
-            raise ValueError("Invalid threshold: all features are discarded.")
+    `threshold_`: float
+        The threshold value used for feature selection.
+    """
 
-        if np.any(mask):
-            mask = safe_mask(X, mask)
-            return X[:, mask]
-        else:
-            raise ValueError("Invalid threshold: all features are discarded.")
+    def __init__(self, estimator, threshold=None, warm_start=False):
+        self.estimator = estimator
+        self.threshold = threshold
+        self.warm_start = warm_start
+
+    def _get_support_mask(self):
+        self.threshold_ = _calculate_threshold(self.estimator, self.scores_,
+                                               self.threshold)
+        return self.scores_ >= self.threshold_
+
+    def fit(self, X, y, **fit_params):
+        """Fit the SelectFromModel meta-transformer.
+
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The training input samples.
+
+        y : array-like, shape = [n_samples]
+            The target values (integers that correspond to classes in
+            classification, real numbers in regression).
+
+        **fit_params : Other estimator specific parameters
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        if not (self.warm_start and hasattr(self, "estimator_")):
```

Comment:
It should be consistent. But it looks to me that it would be without having ``warm_start`` in the ``SelectFromModel`` transformer. It seems a bit odd to me that with the parameter, you need to set ``warm_start`` in both ``SelectFromModel`` and in the model itself.